src/student/components/joinSessionView/StudentFormView.tsx

```diff
@@ -172,7 +174,7 @@ export function StudentFormView(props: StudentFormViewProps) {
                     style={{ width: '100%', maxWidth: "350px" }}
                 />}
             </Backdrop>
-            {!session && (
+            {(!session || session.length !== 7) && (
```

Comment:
It won't do because if you mistype the wrong digit you can't change it once the session variable hits valid length. I've made pull request for your branch that addresses this issue and fixes it #282